static/app/utils/profiling/hooks/useCurrentProjectFromRouteParam.tsx

```diff
@@ -0,0 +1,12 @@
+import {useParams} from 'sentry/utils/useParams';
+import useProjects from 'sentry/utils/useProjects';
+
+export function useCurrentProjectFromRouteParam() {
+  const params = useParams();
+  const projects = useProjects({limit: 1, slugs: [params?.projectId]});
+  const currentProject = projects.projects?.[0] ?? null;
```

Comment:
I would just return on this line, I dont see you using the rest of the projects either (bonus points bc you then get a stable project reference)

Reply:
that `useProjects` hook _could_ make a fetch request and it could be fetched after the fact, this was my only reasoning behind not doing an early return so that i can expose the `projects` obj from `useProjects` in case we ever needed to block or do something with a fetching state.

from what i've been able to test, a project is available to us by the time this event is fired... our usage also is done in such a fashion that we purposely don't block and wait for a project so i could early return 🤷🏼

Reply:
>that useProjects hook could make a fetch request and it could be fetched after the fact, this was my only reasoning behind not doing an early return so that i can expose the projects obj from useProjects in case we ever needed to block or do something with a fetching state.

Fair enough, but since you dont need it right now I would just not add it. The hook will eventually update with the correct value anyways

Reply:
✅